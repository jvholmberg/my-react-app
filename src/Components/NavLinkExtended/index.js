import React from 'react';
import { NavLink } from 'react-router-dom'

const NavLinkExtended = ({ children, exact = true, ...props }) =>
  <NavLink exact { ...props }>{ children }</NavLink>;
export default NavLinkExtended;
