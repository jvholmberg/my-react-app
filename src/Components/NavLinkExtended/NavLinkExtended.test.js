import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import NavLinkExtended from './';

describe('NavLinkExtended', () => {
  it('Only to specified matches snapshot', () => {
    expect(shallow(
      <NavLinkExtended to='/' />
    )).toMatchSnapshot();
  });
});
