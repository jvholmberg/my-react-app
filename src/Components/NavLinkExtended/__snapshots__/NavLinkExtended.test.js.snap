// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinkExtended Only to specified matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavLinkExtended
    to="/"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "ariaCurrent": "true",
      "children": undefined,
      "exact": true,
      "to": "/",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "ariaCurrent": "true",
        "children": undefined,
        "exact": true,
        "to": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
